### PR TITLE
Fix #13521

### LINKED ISSUE #13521: Gluon DataLoader cannot release the processes in the pool
https://github.com/apache/incubator-mxnet/blob/f2dcd7c7b8676b55d912997fc3f9c62c55915307/python/mxnet/gluon/data/dataloader.py#L532-L533

Logically, when a `DataLoader` is recycled, the `_worker_pool` should be recycled, and the `terminate()` of the `_worker_pool` function should be called immediately. However, it did not ... 

Each time I kill a `DataLoader`, it leaves the worker processes dangling.
I guess it is a bug of python `multiprocess.Pool`. Anyway, I think we can patch it by explicitly call `_worker_pool.terminate()`

Minimum code to reproduce the errors.
```python
import mxnet as mx
import numpy as np
A=np.random.rand(999, 2000)
D=mx.gluon.data.DataLoader(A, batch_size=8, num_workers=2)
the_iter = iter(D)
next(the_iter)
del the_iter
del D
```

I recorded a video demo for this bug: https://drive.google.com/open?id=1q4CmU_F1vAtxoZ_KUmrIEfVRk3RsQfv8

Environment: today's mxnet from pip, python3.6 on p3

### PR DESCRIPTION
**Pull request by zhreshold**
## Description ##
Fix #13521 
Python has bug that causing multiprocessing.pool not termniated correctly after gabage collection, this may result in too many dangling processes. 

Fix the problem by manually terminate the pools. 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by zhreshold** [comment]
@eric-haibin-lin Could you help review and merge? Thx
